controllers/openstackbaremetalset_controller.go

```diff
@@ -238,19 +237,127 @@ func (r *OpenStackBaremetalSetReconciler) Reconcile(ctx context.Context, req ctr
 		instance.Status.BaremetalHosts = map[string]ospdirectorv1beta1.OpenStackBaremetalHostStatus{}
 	}
 
+	//
+	// create hostnames for the requested number of systems
+	//
+	//   create new host allocations count
+	//
```

Comment:
I wonder if there's a way to combine this add/delete logic with what we have in `ensureBaremetalHosts` func.  It seems like they do similar things in regards to the IPSet.  That might be tricky though, or perhaps I'm not understanding something.

Reply:
we probably could. we just need to make sure we do the hostname reservation/creation before we to the IP reservation to be able to pass the hostname list. we'd then also need to move the ipset part to ensureBaremetalHosts